controllers/imageclusterinstall_controller.go

```diff
@@ -44,64 +44,80 @@ import (
-	extraManifestsDir          = "extra-manifests"
-	manifestsDir               = "manifests"
-	networkConfigDir           = "network-configuration"
-	clusterConfigFinalizerName = "clusterconfig." + relocationv1alpha1.Group + "/deprovision"
-	caBundleFileName           = "tls-ca-bundle.pem"
+	detachedAnnotation          = "baremetalhost.metal3.io/detached"
+	clusterConfigDir            = "cluster-configuration"
+	extraManifestsDir           = "extra-manifests"
+	manifestsDir                = "manifests"
+	networkConfigDir            = "network-configuration"
+	clusterInstallFinalizerName = "imageclusterinstall." + relocationv1alpha1.Group + "/deprovision"
+	caBundleFileName            = "tls-ca-bundle.pem"
 )
 
 //+kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch
-//+kubebuilder:rbac:groups=relocation.openshift.io,resources=clusterconfigs,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups=relocation.openshift.io,resources=clusterconfigs/status,verbs=get;update;patch
-//+kubebuilder:rbac:groups=relocation.openshift.io,resources=clusterconfigs/finalizers,verbs=update
+//+kubebuilder:rbac:groups=relocation.openshift.io,resources=imageclusterinstalls,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=relocation.openshift.io,resources=imageclusterinstalls/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=relocation.openshift.io,resources=imageclusterinstalls/finalizers,verbs=update
 //+kubebuilder:rbac:groups=metal3.io,resources=baremetalhosts,verbs=get;list;watch;update;patch
 
-func (r *ClusterConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+func (r *ImageClusterInstallReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	log := r.Log.WithFields(logrus.Fields{"name": req.Name, "namespace": req.Namespace})
 	log.Info("Running reconcile ...")
 	defer log.Info("Reconcile complete")
 
-	config := &relocationv1alpha1.ClusterConfig{}
-	if err := r.Get(ctx, req.NamespacedName, config); err != nil {
-		log.WithError(err).Error("failed to get cluster config")
+	clusterInstall := &relocationv1alpha1.ImageClusterInstall{}
+	if err := r.Get(ctx, req.NamespacedName, clusterInstall); err != nil {
+		log.WithError(err).Error("failed to get ImageClusterInstall")
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	if res, stop, err := r.handleFinalizer(ctx, log, config); !res.IsZero() || stop || err != nil {
+	if res, stop, err := r.handleFinalizer(ctx, log, clusterInstall); !res.IsZero() || stop || err != nil {
 		if err != nil {
 			log.Error(err)
 		}
 		return res, err
 	}
 
-	if res, err := r.writeInputData(ctx, log, config); !res.IsZero() || err != nil {
+	if clusterInstall.Spec.ClusterDeploymentRef == nil || clusterInstall.Spec.ClusterDeploymentRef.Name == "" {
+		log.Warn("ClusterDeploymentRef is unset, not reconciling")
+		return ctrl.Result{}, fmt.Errorf("missing ClusterDeploymentRef")
```

Comment:
Is it an error?
I think we can just return, once the CR gets updated with CDRef it will reconcile it again. 
Returning an error will spam the logs because it will get requeued

Reply:
Good point. I'll log an error but not return one